### PR TITLE
Improved Jet Matching

### PR DESCRIPTION
**Pull request by tlprotzman**
Improved jet matching in two ways:

* Limit the eta range to (approximately) detector coverage.  I couldn't quickly find exact ranges for this so for now estimates are used.  This alone greatly improved tracked jet matching
* Fix sign of phi for clustered calo jets.  The sign of phi for truth and reco jets were opposite, leading to very poor matching.  

I also separated EMCal and HCal jets, and added EMCal jets without clustering for debugging.

### CODE REVIEW

**Comment by tlprotzman** [comment]
@raymondEhlers

File: treeAnalysis/caloheader.h [hunk 1]
```diff
@@ -124,7 +123,7 @@ float * CalculateM02andWeightedPosition(std::vector<towersStrct> cluster_towers,
         vecTwr += w_i.at(cellI)*TowerPositionVectorFromIndices(cluster_towers.at(cellI).tower_iEta,cluster_towers.at(cellI).tower_iPhi, caloSelect);
     }
     returnVariables[2]=vecTwr.Eta();
-    returnVariables[3]=(vecTwr.Phi()<0 ? vecTwr.Phi()+TMath::Pi() : vecTwr.Phi()-TMath::Pi());
+    returnVariables[3]=vecTwr.Phi();
```
Comment (tlprotzman) [inline treeAnalysis/caloheader.h:126]: I'm a little concerned about this change; I'm not sure why phi was adjusted in the first place.   However, changing it allows calo jets to match properly.

File: treeAnalysis/caloheader.h [hunk 2]
```diff
@@ -56,7 +56,6 @@ void setBOOLClusterArrayToZero(bool* &arrayinput){
 }
 // conversion functions for processing
 float* EtaPhiFromIndices(int ieta,int iphi,float energy = 0, int caloSelect = 0);
-TVector3 TowerPositionVectorFromIndices(int ieta,int iphi, int caloSelect = 0);
```
Comment (raymondEhlers) [inline treeAnalysis/caloheader.h:59]: Why remove the forward declaration here?

File: treeAnalysis/caloheader.h (hunk 1 above)
Comment (raymondEhlers) [reply treeAnalysis/caloheader.h:126]: I _think_ the idea here was to adjust the phi range, but it looks a bit odd to me. @nschmidtALICE can you comment here?

File: treeAnalysis/treeProcessing.C [hunk 3]
```diff
@@ -587,6 +595,26 @@ void treeProcessing(
                 }
             }
 
+
+            std::vector<double> nocluster_px;
+            std::vector<double> nocluster_py;
+            std::vector<double> nocluster_pz;
+            std::vector<double> nocluster_E;
+            if (true) {
+                for (uint32_t i = 0; i < _nTowers_FEMC; i++) {
```
Comment (raymondEhlers) [inline treeAnalysis/treeProcessing.C:604]: This doesn't matter so much, so you can leave it as is, but I would suggest `std::size_t`

File: treeAnalysis/caloheader.h (hunk 2 above)
Comment (tlprotzman) [reply treeAnalysis/caloheader.h:59]: I don't like to have, but for whatever reason when I added a call to this function [in treeProcessing.C](https://github.com/tlprotzman/AnalysisSoftwareEIC/blob/bdacb5c536b522322334cbb2396bba1c90fcdd29/treeAnalysis/treeProcessing.C#L608) ROOT wasn't running with the declaration, saying it couldn't distinguish between the two call signatures despite it just being a forward declaration.  If you'd like I can investigate it more, honestly I had forgotten about it.

Comment (raymondEhlers) [reply treeAnalysis/caloheader.h:59]: Okay, that's odd, but fair enough - no need to look into it more right now

File: treeAnalysis/caloheader.h (hunk 1 above)
Comment (raymondEhlers) [reply treeAnalysis/caloheader.h:126]: Nico is okay with the change. Merging

**Pull request merged**